libraries/AP_Scripting/docs/docs.lua

```diff
@@ -4202,3 +4202,50 @@ function crsf:get_menu_event(events) end
 ---@param data string -- binary encoded response payload
 ---@return boolean -- true if the repsonse was successfully sent, false otherwise
 function crsf:send_write_response(data) end
+
+-- handle for DroneCAN message operations
+---@class DroneCAN_Handle_ud
+local DroneCAN_Handle_ud = {}
+
+-- create a DroneCAN_Handle, needed for all other DroneCAN message operations
+-- use close() method when finished
+---@param bus_index number -- DroneCAN bus index, 0 for first bus
+---@return DroneCAN_Handle_ud
+function DroneCAN_Handle(bus_index) end
+
+-- return true if last message was used CANFD
+---@return boolean
+function DroneCAN_Handle_ud:canfd() end
```

Comment:
This is probably just me not understanding CANFD, but why would a script care? I thought it was a property of the whole bus and fd messages don't mix with normal ones.

Reply:
If it is per message like this should it not be another return on the `check_message` method?

Reply:
a script cares as the packing with CANFD is (sometimes!) different. In particular we never do TAO with CANFD, so if there is a variable length array at the end of the top level object then the length is encoded in the packet with CANFD and it is not with bxCAN (this stems from the fact that with CANFD you don't know the exact packet length, as the length is rounded up in an odd fashion, so you can't tell the array length reliably from the packet length)

Reply:
I did it as a field for a few reasons. One is to keep as much as possible in auto-generated bindings, another is that nearly everyone won't care about that field (I know they could ignore it), another is that there are other attributes of the transfer that we may also want to add in the future (eg. the arrival timestamp) and I think if we add that then it is cleaner to have these rarely used attributes.
I don't feel strongly about it and I know these are not really strong reasons, so can change if you think it is important

Reply:
Isn't there a race condition problem with this getter if there was a CANFD and a bxCAN node sending the message your subscribed too? How do you match up the fd status to the sending node?

Reply:
fixed